### DISCUSSION TITLE
How to force a specific version of glibc?

### DISCUSSION THREAD

**Discussion by nfachan** [ipetkov/crane#523] [Q&A]
I'd like to force a lower version of glibc for my "official" binaries. How do I go about doing this? I know how to use overlays to specify the rust version, but how is this achieved for glibc?

https://github.com/maelstrom-software/maelstrom/blob/main/flake.nix

**Comment by ipetkov** [comment]
Hi @nfachan ! I'm not an expert on wrangling glibc so take this advice with a grain of salt. I believe the way to achieve this is to construct an `stdenv` which is wired up to use the particular version you are interested in targeting, though there's a few ways to achieve it:

* One way is to [apply an overlay across all of `nixpkgs`](https://discourse.nixos.org/t/compiling-with-old-glibc/12054/4), though the downside is you will likely need to rebuild pretty much everything since newer packages with an older glibc is unlikely to be in the NixOS cache
* Another approach is to [manually construct an `stdenv`](https://discourse.nixos.org/t/including-an-older-glibc-in-stdenv/29478/13) and only use that for your own build (so at least you can reuse the cache for any other dependencies)

Hope this helps!
